### PR TITLE
[ci-app] Improve git Metadata

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?

We attempt to read directly from the `.git` folder, but that sometimes fails: `git-repo-info` (the library we use to read) does not account for all the possible ways the commit info is encoded. 

What this PR does is to add a fallback to the `git` command if we can't successfully read from `.git` ourselves. 

### Motivation
Increase the odds of having `git` metadata.

### Plugin Checklist

- [x] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: packages/dd-trace/src/plugins/util/git.js [hunk 1]
```diff
@@ -29,39 +29,81 @@ function parseUser (user) {
...
     [GIT_COMMIT_COMMITTER_DATE]: committerDate,
     [GIT_COMMIT_COMMITTER_NAME]: committerName,
-    [GIT_COMMIT_COMMITTER_EMAIL]: committerEmail
+    [GIT_COMMIT_COMMITTER_EMAIL]: committerEmail,
+    [GIT_BRANCH]: coalesce(branch, gitBranch),
+    [GIT_COMMIT_SHA]: coalesce(commitSHA, gitCommitSHA),
+    [GIT_TAG]: coalesce(tag, gitTag)
   }
+
+  return removeEmptyValues(spanTags)
```
Comment (rochdev) [inline packages/dd-trace/src/plugins/util/git.js:106]: Why is this needed? The tracer already ignores tags that are `null` or `undefined`. If the value here is an empty string, I think it would be safe to update the tracer to always ignore empty strings as well.

Comment (juan-fernandez) [reply packages/dd-trace/src/plugins/util/git.js:106]: that makes sense: where would that need to be made?

Comment (rochdev) [reply packages/dd-trace/src/plugins/util/git.js:106]: That would be in `format.js` since it's the file responsible for filtering out empty tags. More specifically: https://github.com/DataDog/dd-trace-js/blob/ea6c469abb02166a0328bf8aa2a007881ff61b2d/packages/dd-trace/src/format.js#L140-L166

Comment (juan-fernandez) [reply packages/dd-trace/src/plugins/util/git.js]: done in https://github.com/DataDog/dd-trace-js/pull/1360/commits/ecffdd5be31d310edc73d9cf49eadd4f7831c85f

**Review by rochdev — APPROVED** [review]
